test/functional/options/defaults_spec.lua

```diff
@@ -616,10 +616,33 @@ describe('stdpath()', function()
 
     -- Check that Nvim rejects invalid APPNAMEs
     -- Call jobstart() and jobwait() in the same RPC request to reduce flakiness.
-    eq(1, exec_lua([[
-      local child = vim.fn.jobstart({ vim.v.progpath }, { env = { NVIM_APPNAME = 'a/b\\c' } })
-      return vim.fn.jobwait({ child }, 3000)[1]
-    ]]))
+    local function test_appname(testAppname, expectedExitCode)
+      -- wait for less time if the expectation is success
+      local waitTime = (expectedExitCode == -1) and 100 or 3000
+      local lua_code = string.format([[
+        local child = vim.fn.jobstart({ vim.v.progpath }, { env = { NVIM_APPNAME = %q } })
+        return vim.fn.jobwait({ child }, %d)[1]
+      ]], alter_slashes(testAppname), waitTime)
+
+      eq(expectedExitCode, exec_lua(lua_code))
+    end
+    -- APPNAMEs that should be rejected
+    test_appname('a/../b', 1)
+    test_appname('../a', 1)
+    test_appname('a/..', 1)
+    test_appname('..', 1)
+    test_appname('.', 1)
+    test_appname(is_os('win') and 'C:/a/b' or '/a/b', 1)
+    -- APPNAMEs that should be accepted
+    test_appname('a/b', -1)
+    test_appname('a/b/c', -1)
+  end)
+
+  it('tempname() succeeds when $NVIM_APPNAME is a path', function()
+      clear({env={
+        NVIM_APPNAME=alter_slashes('a/b'),
+      }})
+      funcs.tempname() -- should not crash
```

Comment:
use `assert_alive()` after this, or ideally check for the actual value (the above test already has an approach to do that)

Reply:
The thing that's throwing me with this is that my test gives it `/home/rory/Documents/projects/neovim/build/Xtest_tmpdir/a%b.rory/J4qGVP/0`
Is there a variable for `Xtest_tmpdir` that I just can't find?

Reply:
Okay so I fixed that by setting `TMPDIR`, but setting `USER` did nothing, I for sure thought `USER` was the obvious one
I'll try fix it in a bit